File: cpp/merge_strings_alternately.cpp
#include <iostream>

using namespace std;

class Solution{
  public:
    string mergeStrings(string str1, string str2){
      int l1 = str1.size(), l2 = str2.size();
      int i = 0;
      string res = "";
      while(i < l1 || i < l2){
        if(i < l1) res += str1[i];
        if(i < l2) res += str2[i];
        ++i;
      }
      return res;
    }
};

int main(){
  Solution s;
  cout << s.mergeStrings("abcde", "pqr") << endl;
  cout << s.mergeStrings("abc", "pqrs") << endl;
  cout << s.mergeStrings("leet", "code") << endl;
}

File: java/Array-Strig/MergeStringsAlternately.java
public class MergeStringsAlternately{
  public static void main(String[] args){
    Solution s = new Solution();
    System.out.println(s.mergedString("abc", "pqrs"));
    System.out.println(s.mergedString("abcde", "pqr"));
    System.out.println(s.mergedString("", ""));
  }
}

class Solution{
  public String mergedString(String str1, String str2){
    StringBuilder res = new StringBuilder();
    int l1 = str1.length(), l2 = str2.length();
    char[] ch1 = str1.toCharArray();
    char[] ch2 = str2.toCharArray();
    int i = 0;
    while(i < l1 || i < l2){
      if(i < l1) res.append(ch1[i]);
      if(i < l2) res.append(ch2[i]);
      ++i;
    }
    return res.toString();
  }
}